contracts/core/native_fallback/L1/Registry.sol

```diff
@@ -275,16 +308,26 @@ contract Registry is IRegistry, Ownable, Pausable, AccessControl {
      * @return bool True if the address is a grantee for the chain ID, false otherwise
      */
     function _isGrantee(address _grantee, uint256 _chainID) internal view returns (bool) {
-        bytes32 role = _getChainRole(_chainID);
-        return hasRole(role, _grantee);
+        return hasRole(_getChainRole(_chainID, false), _grantee);
+    }
+
+    /**
+     * @dev Internal function to check if an address is a grantee for a specific chain ID
+     * @param _grantee The address to check
+     * @param _chainID The chain ID to check for
+     * @return bool True if the address is a grantee for the chain ID, false otherwise
+     */
+    function _isIrrevocableGrantee(address _grantee, uint256 _chainID) internal view returns (bool) {
+        return hasRole(_getChainRole(_chainID, true), _grantee);
     }
 
     /**
      * @dev Get chain-specific role identifier based on chain ID
      * @param _chainID The chain ID to get the role for
+     * @param irrevocable The chain ID to get the role for
      * @return bytes32 The role identifier for the chain ID
      */
-    function _getChainRole(uint256 _chainID) internal pure returns (bytes32) {
-        return keccak256(abi.encode(_CHAIN_ROLE_PREFIX, _chainID));
+    function _getChainRole(uint256 _chainID, bool irrevocable) internal pure returns (bytes32) {
```

Comment:
Key change here is that we irrevocability results in a different role, as it should.